server/service/src/sync/translations/mod.rs

```diff
@@ -199,93 +199,107 @@ pub(crate) trait SyncTranslation {
         Ok(None)
     }
 
-    fn try_translate_push(
+    /// Implementation should return three types of results
+    /// * Error - Something completely unexpected that is not recoverable
+    /// * None - Translator did not match record type
+    /// * Some - Translator did match and either translated record/records or
+    ///          empty array if record is deliberatly ignored
+    fn try_translate_push_upsert(
         &self,
         _: &StorageConnection,
         _: &ChangelogRow,
-    ) -> Result<Option<Vec<PushUpsertRecord>>, anyhow::Error> {
+    ) -> Result<Option<Vec<RemoteSyncRecordV5>>, anyhow::Error> {
         Ok(None)
     }
-}
 
-#[derive(Debug)]
-pub(crate) struct PushUpsertRecord {
-    pub(crate) sync_id: i64,
-    /// The translated table name
-    pub(crate) table_name: &'static str,
-    pub(crate) record_id: String,
-    pub(crate) data: serde_json::Value,
+    fn try_translate_push_delete(
+        &self,
+        _: &StorageConnection,
+        _: &ChangelogRow,
+    ) -> Result<Option<Vec<RemoteSyncRecordV5>>, anyhow::Error> {
+        Ok(None)
+    }
 }
 
-pub(crate) struct PushDeleteRecord {
-    pub(crate) sync_id: i64,
-    /// The translated table name
-    pub(crate) table_name: &'static str,
-    pub(crate) record_id: String,
+impl RemoteSyncRecordV5 {
```

Comment:
I like this associated class methods, I think this is a good example of it being used